diff -Naru src-orig/handler.cc src-mod/handler.cc
--- src-orig/handler.cc	2019-02-16 16:10:30.158007582 +0200
+++ src-mod/handler.cc	2019-02-16 16:32:26.005219510 +0200
@@ -290,7 +290,8 @@
                     {
 
                         if (  prefs.extra_buttons.ref().size() && (guint) event->detail == prefs.extra_buttons.ref().rbegin()->button) //check if the button is same as last extra button
-                            event->detail = prefs.button.ref().button; //fake the default button
+                            event->detail = 100+event->detail; //fake the default button
+                            //event->detail = prefs.button.ref().button; //fake the default button
                     }
 
 			H->release(event->detail, create_triple(event->root_x, event->root_y, event->time));
@@ -980,7 +981,11 @@
 		RStroke s = finish(0);
 
         if (experimental) //button 1 should be treated as trigger 0, other trigger buttons copied from the event button b itself
-            s->trigger = b>1 ? b : 0 ; 
+            if (b>100) 
+            {
+                b=b-100;
+                s->trigger = prefs.button.ref().button-1; 
+            }
         
 		if (prefs.move_back.get() && !xstate->current_dev->absolute)
 			XTestFakeMotionEvent(dpy, DefaultScreen(dpy), orig->x, orig->y, 0);
diff -Naru src-orig/handler.cc.orig src-mod/handler.cc.orig
--- src-orig/handler.cc.orig	2019-02-16 16:10:30.158007582 +0200
+++ src-mod/handler.cc.orig	2013-03-27 17:52:38.000000000 +0200
@@ -281,16 +281,6 @@
 				break;
 			xinput_pressed.erase(event->detail);
 			in_proximity = get_axis(event->valuators, current_dev->proximity_axis);
-
-            if (experimental) //if device is disabled, but extra buttons followed, treat last extra button as the default button
-                for (std::set<std::string>::iterator i = prefs.excluded_devices.ref().begin(); i != prefs.excluded_devices.ref().end(); i++)
-                    if (! i->compare(grabber->get_xi_dev(event->deviceid)->name) ) //check if the grabbed device name is in disabled device list
-                    {
-
-                        if (  prefs.extra_buttons.ref().size() && (guint) event->detail == prefs.extra_buttons.ref().rbegin()->button) //check if the button is same as last extra button
-                            event->detail = prefs.button.ref().button; //fake the default button
-                    }
-
 			H->release(event->detail, create_triple(event->root_x, event->root_y, event->time));
 			break;
 		case XI_Motion:
@@ -979,9 +969,6 @@
 	virtual void release(guint b, RTriple e) {
 		RStroke s = finish(0);
 
-        if (experimental) //button 1 should be treated as trigger 0, other trigger buttons copied from the event button b itself
-            s->trigger = b>1 ? b : 0 ; 
-        
 		if (prefs.move_back.get() && !xstate->current_dev->absolute)
 			XTestFakeMotionEvent(dpy, DefaultScreen(dpy), orig->x, orig->y, 0);
 		else
